### DISCUSSION TITLE
How to calculate the mass and stiffness of 20 tower nodes？

### DISCUSSION THREAD

**Discussion by icnosvbovwb** [OpenFAST/openfast#1502] [Q&A]
Dear @jjonkman 
I obtain the TMassDen FAStiff and SStiff of 20 tower nodes through the. ed file. Now I want to calculate the mass and stiffness of the 20 nodes. The mass can be achieved through mass=DHNodes*TMassDen，Right？ 
Similarly, The stiffness=FAStiff/HNodes/HNodes/HNodes, right?
![image](https://user-images.githubusercontent.com/107483730/226898212-ef86a1dc-23c0-4a25-8b87-44ea21f9674a.png)

**Comment by jjonkman** [comment]
Dear @icnosvbovwb,

I agree that the mass of a tower element in ElastoDyn is `DHNodes*TMassDen`.

But I'm not sure what stiffness you are trying to calculate.  Are you wanting the generalized stiffness matrix of the bending modes of the tower, which will depend on the distributed stiffness (`FAStiff`, `SSStiff`), as well as on the curvature of the user-specified mode shapes?

Best regards,

**Reply by icnosvbovwb** [reply]
Dear @jjonkman 
I want to calculate the mode shapes and frequency of each node based on the mass and stiffness of each node，such like [V,d]=eig(K,M); Therefore, this stiffness is not bending stiffness, so I am not sure how to calculate stiffness, In other words, the stiffness I want is the stiffness in N/m

**Reply by jjonkman** [reply]
Dear @icnosvbovwb,

Just a couple comments:
- The mode shapes of the tower are inputs to ElastoDyn, not outputs from ElastoDyn.  It sounds like you want to know how to calculate these mode shapes?  You can use software such as BModes_JJ for that, as discussed many times on our forum: https://forums.nrel.gov/.
- When calculating the mode shapes of continuous beams, it is the mass per unit length (`TMassDen`) and cross-sectional stiffness you need to use (EI, which is `FAStiff` and `SSStiff`) in Nm^2 at each cross section, not a spring stiffness in N/m.

Best regards,

**Reply by icnosvbovwb** [reply]
Dear @jjonkman 
Thanks for your reply. If I use Bmodes to calculate modal shapes, I find only two examples（CS_Monopile.bmi and OC3Hywind.bmi）. 
Q1:How can I calculate OC4 or other wind turbine examples? 
Q2:I tried to calculate the OC3Hywind.bmi example, and the results obtained are basically consistent with the frequency of free attenuation I tested, but the tower is slightly different. Do you think the results are correct?
Q3:Modes 1 to 2 correspond to sway and surge, mode 3 corresponds to heave, modes 4 to 5 correspond to roll and pitch, and mode 6 corresponds to yaw. Mode 7 and later modes are the tower mode. Right?
![image](https://user-images.githubusercontent.com/107483730/226957677-e02ccc1d-b4a8-4c72-a509-0fc043200ec5.png)
![image](https://user-images.githubusercontent.com/107483730/226958874-ae6a106f-c720-471f-880d-ddc24e11ae0d.png)

**Reply by jjonkman** [reply]
Dear @icnosvbovwb,

Here are my responses:
1. You can follow the approach used in these examples and apply them to other floating support structures.  For the OC4-DeepCwind semisubmersible, I would start with the OC3-Hywind example and change the floater properties as necessary.
2. Yes, the results look correct to me.  There are some differences in the theory basis of BModes_JJ and OpenFAST (which is what I presume you used to derive the free-decay results) that will lead to small differences in the predicted tower frequencies between the two software, but the mode shapes derived from BModes_JJ should be accurate enough for use in ElastoDyn.
3. Correct.

Best regards,
